pkgs/development/libraries/arrayfire/default.nix

```diff
@@ -1,91 +1,177 @@
-{ lib
-, stdenv
-, fetchFromGitHub
+{ blas
+, boost
 , cmake
-, pkg-config
-, opencl-clhpp
-, ocl-icd
+, config
+, cudaPackages
+, fetchFromGitHub
 , fftw
 , fftwFloat
-, blas
+, fmt_9
+, forge
+, freeimage
+, git
+, gtest
 , lapack
-, boost
-, mesa
-, libGLU
+, lib
 , libGL
-, freeimage
+, mesa
+, ocl-icd
+, opencl-clhpp
+, pkg-config
 , python3
-, clfft
-, clblas
-, doxygen
-, buildDocs ? false
-, config
-, cudaSupport ? config.cudaSupport
-, cudatoolkit
-, darwin
+, span-lite
+, stdenv
+, doCheck ? false
+, withCPU ? true
+, withCuda ? config.cudaSupport
+, withOpenCL ? stdenv.isLinux
+, nvidiaComputeDrivers ? null
 }:
 
+assert blas.isILP64 == false;
+
 stdenv.mkDerivation rec {
   pname = "arrayfire";
-  version = "3.7.3";
+  version = "3.9.0-pre";
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
-    rev = "v${version}";
-    sha256 = "0gcbg6b6gs38xhks5pp0vkcqs89zl7rh9982jqlzsd0h724qddw0";
-    fetchSubmodules = true;
+    rev = "d2a66367d859cdb554f2374e29d39c88d5fff978";
+    hash = "sha256-9r1w0U9MvhduHwBpEWpqkrQPawd94EY3FAqSJghi09I=";
+  };
+
+  assets = fetchFromGitHub {
+    owner = pname;
+    repo = "assets";
+    rev = "cd08d749611b324012555ad6f23fd76c5465bd6c";
+    sha256 = "sha256-v4uhqPz1P1g1430FTmMp22xJS50bb5hZTeEX49GgMWg=";
+  };
```

Comment:
A comment explaining what these are, "Assets for the ArrayFire documentation". I also wonder if there's a flag allowing to opt out of building the documentation (and thus out of carrying these PNGs around at runtime)

Reply:
I managed to get rid of these altogether.